### PR TITLE
Add functional tests on arm64

### PR DESCRIPTION
**Pull request by dims**
Signed-off-by: Davanum Srinivas <davanum@gmail.com>

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

**Comment by dims** [comment]
Removing WIP, the CI job is green

image

**Comment by dims** [comment]
/assign @serathius @spzala 

cc @chaochn47 @geetasg Geeta, Chao, can you please follow the same pattern and stand up other CI jobs needed for etcd to claim `Tier 1` support for `arm64`?

**Review by spzala — APPROVED** [review]
lgtm, thank you so much @dims @serathius

File: .github/workflows/functional-arm64.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: functional-arm64-tests
+on: [push, pull_request]
+permissions: read-all
+jobs:
+  test:
+    runs-on: [Linux, ARM64]
```
Comment (ahrtr) [inline .github/workflows/functional-arm64.yaml:6]: This seems not correct to me. `Linux` is `GOOS` type, while `ARM64` is `GOARCH` type.

Comment (dims) [reply .github/workflows/functional-arm64.yaml:6]: @ahrtr these are tags you see on the GH self hosted runner.

image

File: .github/workflows/functional-arm64.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        with:
+          go-version: "1.19.5"
+      - run: date
+      - env:
+          TARGET: ${{ matrix.target }}
+        run: |
+          echo "${TARGET}"
+          case "${TARGET}" in
+            linux-arm64-functional)
+              GO_BUILD_FLAGS='-v -mod=readonly' ./scripts/build.sh && GOARCH=arm64 PASSES='functional' ./scripts/test.sh
```
Comment (ahrtr) [inline .github/workflows/functional-arm64.yaml:24]: We are going to deprecate the functional test, FYI. https://github.com/etcd-io/etcd/issues/15102

If we really need to promote arm64 to tier 1 support, I would suggest to extend the all or majorities of existing workflow to cover `ARM64`. cc @ptabor @serathius @spzala

Comment (dims) [reply .github/workflows/functional-arm64.yaml:24]: @ahrtr of course! I am requesting @geetasg and @chaochn47 who will be helping out to identify and add what ever jobs needed.

File: .github/workflows/functional-arm64.yaml (hunk 1 above)
Comment (ahrtr) [reply .github/workflows/functional-arm64.yaml:6]: Got it, it's a combination of the tags. 

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idruns-on

Comment (ahrtr) [reply .github/workflows/functional-arm64.yaml:6]: thx @dims , please see my another comment below.

Comment (dims) [reply .github/workflows/functional-arm64.yaml:6]: yep done :)

File: .github/workflows/functional-arm64.yaml (hunk 2 above)
Comment (ahrtr) [reply .github/workflows/functional-arm64.yaml:24]: Sure, it's OK to do it in a separate PR.

**Review by ahrtr — APPROVED** [review]
LGTM

Thank you @dims 

cc  @ptabor @serathius for awareness and approval as well.

**Comment by serathius** [comment]
Could we begin with arm64 tests marked as optional? With only 2 machines running through multiple workflows and multiple parallel PRs might substantially increase execution time.

**Comment by dims** [comment]
> Could we begin with arm64 tests marked as optional? With only 2 machines running through multiple workflows and multiple parallel PRs might substantially increase execution time.

@serathius turned it into a cron job. is that ok?

**Comment by serathius** [comment]
Looks good.

**Review by serathius — APPROVED** [review]

**Pull request merged**